### PR TITLE
[serverless] fix IAM settings

### PR DESCRIPTION
**Pull request by Elhebert**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - [serverless.yml reference](https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/serverless.yml.md)
  - [serverless provider.iam changes](https://www.serverless.com/framework/docs/deprecations#grouping-iam-settings-under-provideriam)

### CODE REVIEW

**Review by G-Rath — CHANGES_REQUESTED** [review]
Could you please update the tests to reflect your changes :)

File: types/serverless/plugins/aws/provider/awsProvider.d.ts [hunk 1]
```diff
@@ -81,11 +81,16 @@ declare namespace Aws {
     }
 
     interface IamSettings {
-        role?: string;
+        role?: string | IamLogicalRole;
+        deploymentRole?: string;
+    }
+    
+    interface IamLogicalRole {
```
Comment (G-Rath) [inline types/serverless/plugins/aws/provider/awsProvider.d.ts:88]: Any reason why you've called this `IamLogicalRole` instead of just `IamRole`?

Comment (Elhebert) [reply types/serverless/plugins/aws/provider/awsProvider.d.ts:88]: Not really. 

When I looked at the serverless.yml reference given by Serverless they have a comment that seems to call this a logical role. I can change this for `IamRole` as it does make more sense.

serverless.yml reference:
```yml
  iam:
    # Overwrite the default IAM role which is used for all functions
    role: arn:aws:iam::XXXXXX:role/role
    # .. OR configure logical role
    role:
```

**Review by G-Rath — APPROVED** [review]

**Comment by Elhebert** [comment]
Ready to merge

**Pull request merged**